src/Microsoft.ML.Data/Scorers/FeatureContributionCalculatingTransformer.cs

```diff
@@ -41,88 +49,192 @@ namespace Microsoft.ML.Runtime.Data
     /// ]]>
     /// </format>
     /// </example>
-    public sealed class FeatureContributionCalculationTransform
+    public sealed class FeatureContributionCalculatingTransformer : RowToRowTransformerBase
     {
         // Apparently, loader signature is limited in length to 24 characters.
-        internal const string MapperLoaderSignature = "WTFBindable";
-        private const int MaxTopBottom = 1000;
+        internal const string Summary = "For each data point, calculates the contribution of individual features to the model prediction.";
+        internal const string FriendlyName = "Feature Contribution Transform";
+        internal const string LoaderSignature = "FeatureContribution";
 
-        public sealed class Arguments : ScorerArgumentsBase
-        {
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of top contributions", SortOrder = 1)]
-            public int Top = 10;
+        private readonly string _featureColumn;
+        private readonly BindableMapper _mapper;
 
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Number of bottom contributions", SortOrder = 2)]
-            public int Bottom = 10;
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
```

Comment:
Do we preserve backwards compatibility with `What The Feature`? We need to support the old signatures.

Reply:
@ivmatan@microsoft.com I am not sure what backward compatibility means here, could you tell me what I need to preserve?

---
In reply to: [239965920](https://github.com/dotnet/machinelearning/pull/1847#discussion_r239965920) [](ancestors = 239965920)

Reply:
That means if you take model trained before you changes and try to load, you would be able to load it fine.
One thing is to check Save methods, and what you have same format.
You can also test it, see TestSelectBackCompatChooseColumnsWithKeep

---
In reply to: [241156251](https://github.com/dotnet/machinelearning/pull/1847#discussion_r241156251) [](ancestors = 241156251,239965920)